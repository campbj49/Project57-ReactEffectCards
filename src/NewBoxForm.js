import { useState } from "react";
import "./NewBoxForm.css"
/**
 * NewBoxForm: Creates and handles the form for making a new box
 * 
 * Props: 
 * -onSubmit: function for passing the form data up to the parent
 * 
 * States:
 * -height: current value typed into the form
 * -width:  "                "             "
 * -color:  "                "             "
 * 
 * App --> BoxList --> NewBoxForm
 */

function NewBoxForm({onSubmit}){
    //data and handlChange() modified from example
    const [formData, setFormData] = useState({
        height: "",
        width: "",
        backgroundColor: ""
      });
    //keeps input val props up to date
    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    return(
        <form onSubmit={onSubmit}>
            <label htmlFor="height">Height</label>
            <input required type="number" id="height" onChange={handleChange}></input><br/>
            <label htmlFor="width">Width</label>
            <input required type="number"id="width" onChange={handleChange}></input><br/>
            <label htmlFor="backgroundColor">Background Color</label>
            <input required type="text" id="backgroundColor" onChange={handleChange}></input><br/>
            <button>Create Box</button>
        </form>
    )
}

export default NewBoxForm;