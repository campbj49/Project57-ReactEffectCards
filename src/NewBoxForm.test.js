import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";
import TEST_IMAGES from "./_testCommon.js";

//smoke test to make sure it renders in the first place from smoke test example
it("renders without crashing", function() {
  render(<NewBoxForm/>);
});

// snapshot test from snapshot example
it("matches snapshot", function() {
  const {asFragment} = render(<NewBoxForm/>);
  expect(asFragment()).toMatchSnapshot();
});