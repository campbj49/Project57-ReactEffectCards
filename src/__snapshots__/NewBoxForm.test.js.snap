// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      required=""
      type="number"
    />
    <br />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      required=""
      type="number"
    />
    <br />
    <label
      for="backgroundColor"
    >
      Background Color
    </label>
    <input
      id="backgroundColor"
      required=""
      type="text"
    />
    <br />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
